### PR TITLE
👷(github) create pull request for richie-site-factory after tagging

### PR DESCRIPTION
**Pull request by Sifflex**
## Purpose

Since tagging is relatively frequent, we improve the conveniency of
releasing a version of richie by automatically creating a pull request
updating all richie-related sites

## Proposal

We added a github action that is triggered everytime a new tag is created on the repository which launches a python script to create a pull request to update richie-site-factory

### CODE REVIEW

File: .github/workflows/deploy_richie.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so the job can access it
+      - uses: actions/checkout@v2
+
+      - name: Install Dependencies
+        run: pip3 install setuptools; pip3 install GitPython; pip3 install PyGithub;
+
+      # GITHUB_REF##*/ turns refs/heads/v1.3 to v1.3
+      - name: Deploy
```
Comment (sampaccoud) [inline .github/workflows/deploy_richie.yml:20]: It's not a deployment, more an upgrade!

File: .github/workflows/deploy_richie.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+commits = cmd.run(
+    f'cd {repo.name}; \
+                    git log --pretty="%s" {branch_name}',
+    stdout=cmd.PIPE,
+    shell=True,
+).stdout.decode("utf-8")
+last_commits = commits.split("\n")[:number_of_sites]
+
+body = ""
+for commit in last_commits:
```
Comment (sampaccoud) [inline .github/workflows/deploy_richie.py:91]: this is done by the bin/upgrade script if you pass the -c option.

Comment (Sifflex) [reply .github/workflows/deploy_richie.py:91]: The body is the one used to create the pull request because empty bodies are not allowed

**Comment by sampaccoud** [comment]
github.com/openfun/richie-site-factory was deprecated and the site factory brought into the present repository with Cookiecutter.

**Pull request closed without merging**